### PR TITLE
chore: Update @tinymce/tinymce-react

### PR DESCRIPTION
**Pull request by sbalaji1192**
## Automation

/ok-to-test tags="@tag.TextEditor"

### :mag: Cypress test results

> [!IMPORTANT]
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/8477514930>
> Commit: `b69e1d01c55200b58eea0704e36b940bbb8f1992`
> Cypress dashboard url: <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=8477514930&attempt=1" target="_blank">Click here!</a>
> All cypress tests have passed 🎉🎉🎉

## Summary by CodeRabbit

- **New Features**
	- Enhanced Rich Text Editor with improved styling, additional toolbar options, and updated functionality.
- **Style**
	- Updated styles for the Rich Text Editor, including toolbar, editor header, and various elements within the editor.
- **Chores**
	- Removed unused plugins from the Rich Text Editor.

### CODE REVIEW

File: app/client/src/widgets/RichTextEditorWidget/component/index.tsx [hunk 1]
```diff
@@ -168,7 +169,7 @@ const StyledRTEditor = styled.div<{
   }
 
   .tox-editor-header {
-    border-bottom: 1px solid var(--wds-color-border);
+    border-bottom: 1px solid var(--wds-color-border) !important;
```
Comment (KelvinOm) [inline app/client/src/widgets/RichTextEditorWidget/component/index.tsx:172]: Can we avoid `important ` here?

Comment (sbalaji1192) [reply app/client/src/widgets/RichTextEditorWidget/component/index.tsx]: Removed,

File: app/client/src/widgets/RichTextEditorWidget/component/index.tsx [hunk 2]
```diff
@@ -237,6 +239,44 @@ const StyledRTEditor = styled.div<{
...
+  .tox-statusbar__help-text {
+    display: none;
+  }
+
+  .tox .tox-tbtn--bespoke {
+    background: #fff;
+  }
+`;
+
+const GlobalStyles = createGlobalStyle`
```
Comment (KelvinOm) [inline app/client/src/widgets/RichTextEditorWidget/component/index.tsx:259]: Why should it be global styles?

**Review by KelvinOm — APPROVED** [review]
LGTM! Pay attention to the failed E2E tests.

File: app/client/src/widgets/RichTextEditorWidget/component/index.tsx (hunk 2 above)
Comment (sbalaji1192) [reply app/client/src/widgets/RichTextEditorWidget/component/index.tsx:259]: @KelvinOm This is for the popups that appear outside the widget subtree.

**Review by somangshu — APPROVED** [review]
LGTM
Sanity checked the RTE widget

**Pull request merged**